### PR TITLE
fix: [CDS-72695]: made description and short description field as non-mandatory

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowUpdate/ServiceNowUpdateStepMode.tsx [hunk 1]
```diff
@@ -728,14 +728,6 @@ function ServiceNowUpdateStepMode(
             requiredErrorMsg: getString('pipeline.serviceNowApprovalStep.validations.connectorRef')
           }),
           ticketType: Yup.string().required(getString('pipeline.serviceNowApprovalStep.validations.ticketType')),
-          description: Yup.string().when('useServiceNowTemplate', {
-            is: false,
-            then: Yup.string().required(getString('pipeline.serviceNowCreateStep.validations.description'))
-          }),
-          shortDescription: Yup.string().when('useServiceNowTemplate', {
-            is: false,
-            then: Yup.string().required(getString('pipeline.serviceNowCreateStep.validations.shortDescription'))
```
Comment (sonaligoyal-harness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowUpdate/ServiceNowUpdateStepMode.tsx:737]: There are validations in deployment form also for these fields. Can you please check that in ServiceNowUpdate file and remove from there too?

Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowUpdate/ServiceNowUpdateStepMode.tsx:737]: done, pls re-review

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx [hunk 2]
```diff
@@ -709,14 +709,6 @@ function ServiceNowCreateStepMode(
...
           }),
           ticketType: Yup.string().required(getString('pipeline.serviceNowApprovalStep.validations.ticketType')),
-          description: Yup.string().when('useServiceNowTemplate', {
-            is: false,
-            then: Yup.string().required(getString('pipeline.serviceNowCreateStep.validations.description'))
-          }),
-          shortDescription: Yup.string().when('useServiceNowTemplate', {
-            is: false,
-            then: Yup.string().required(getString('pipeline.serviceNowCreateStep.validations.shortDescription'))
-          }),
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx:719]: removing code always feels good

**Review by ishantmehtaharness — DISMISSED** [review]
looks good

**Comment by ishantmehtaharness** [comment]
checks are failing

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx (hunk 2 above)
Comment (Sagar-Rajput7) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx:719]: agreed 😅

**Review by vikrant-harness — DISMISSED** [review]

**Review by Sagar-Rajput7 — DISMISSED** [review]

**Review by PratyushGar — DISMISSED** [review]

**Review by Eric761 — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Comment by ishantmehtaharness** [comment]
looks good

**Pull request merged**